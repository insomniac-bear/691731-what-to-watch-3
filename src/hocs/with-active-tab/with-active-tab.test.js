import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import withActiveTab from './with-active-tab.js';

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withActiveTab(MockComponent);

it(`withVideo is renderered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      activeTab={`Active tab`}
    />
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
