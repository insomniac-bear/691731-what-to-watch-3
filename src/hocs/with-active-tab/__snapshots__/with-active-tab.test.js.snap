// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withVideo is renderered correctly 1`] = `<div />`;
